sarpy/io/phase_history/cphd.py

```diff
@@ -1376,20 +1418,20 @@ def _check_fully_written(self):
 
         if self.cphd_meta.Data.SupportArrays is not None:
             for entry in self.cphd_meta.Data.SupportArrays:
-                status = False
                 support_pixels = entry.NumRows*entry.NumCols
                 if self._writing_state['support'][entry.Identifier] < support_pixels:
+                    status = False
                     support_message += 'identifier {}, {} of {} pixels written\n'.format(
                         entry.Identifier, self._writing_state['support'][entry.Identifier], support_pixels)
 
         if not status:
-            logging.error('CPHD file {} is not completely written, and the result may be corrupt.'.format(self._file_name))
+            logging.error('CPHD file %s is not completely written, and the result may be corrupt.', self._file_name)
```

Comment:
I don't mind, but why should this version of string formatting be preferable?

Reply:
Sorry, I should've called more attention to this in the description. Depending on your preferences, [pylint flags this](https://vald-phoenix.github.io/pylint-errors/plerr/errors/logging/W1202.html). I think the idea is that when called in this manner, the string formatting can be lazy and only produce a complete log message if the message is actually getting logged somewhere.

Reply:
I've never had an issue with this, but it's good to know. I seldom log anything at a frequency that the difference between lazy and non-lazy logging would actually matter. I asked because I assumed that there must be some rationale. Thanks.